streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -46,43 +46,90 @@
  * @param <K>
  * @param <V>
  */
-public class MeteredKeyValueStore<K, V>
-    extends WrappedStateStore<KeyValueStore<Bytes, byte[]>, K, V>
+public class MeteredKeyValueStore<K, V, OwnType extends MeteredKeyValueStore<K, V, OwnType>>
```

Comment:
The self-type reference is pretty mind-bending, but basically what it's for is to let `reWrap` return the same type you call it on. So, `MeteredKeyValueStore#reWrap` returns a `MeteredKeyValueStore` with the same generic types, and `MeteredTimestampedKeyValueStore#reWrap` returns a `MeteredTimestampedKeyValueStore` with the same generic parameters.